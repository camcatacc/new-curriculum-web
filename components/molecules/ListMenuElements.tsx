// Modules
import React from "react";

// Elements
import { MenuElement } from "components/atoms/MenuElement/MenuElement";

// Definitions
export interface ListMenuElementsProps {
	menuElementsNames: string[];
}

// Element
const ListMenuElements = ({ menuElementsNames }: ListMenuElementsProps) => (
	<div className="flex-row flex gap-28">
		{menuElementsNames.map((val, ind) => (
			<MenuElement title={val} onClicked={() => {}} selected={false} key={`menuElement${ind}`} />
		))}
	</div>
);
export default ListMenuElements;
