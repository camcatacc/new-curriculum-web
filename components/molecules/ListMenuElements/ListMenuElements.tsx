// Modules
import React, { useEffect, useState } from "react";

// Elements
import MenuElement from "components/atoms/MenuElement/MenuElement";

// Definitions
export interface ListMenuElementsProps {
	menuElements: { name: string; id: string; onClick: () => void }[];
	selectedId: string;
	className?: string;
}

/* const MenuContainer = ({ className, children, isMobile }: { className?: string; children?: false | JSX.Element[]; isMobile?: boolean }) => {
	const [open, setOpen] = useState(false);

	return isMobile ? (
		<div>
			<IconButton onClick={() => setOpen(!open)} color="primary">
				<MoreVert />
			</IconButton>
			<Drawer open={open} onClose={() => setOpen(false)}>
				{children}
			</Drawer>
		</div>
	) : (
		<div className={`${className} flex-row flex gap-28`}>{children}</div>
	);
};*/

// Element
const ListMenuElements = ({ menuElements, selectedId, className }: ListMenuElementsProps) => {
	const [isMobile, setIsMobile] = useState(false);
	useEffect(() => {
		const media = window.matchMedia(`(max-width: 600px)`);
		media.addEventListener("change", (e) => isMobile !== e.matches && setIsMobile(e.matches));
	});

	if (isMobile) return <></>;

	return (
		//	<MenuContainer isMobile={isMobile} className={className}>
		<div className={`${className} flex-row flex gap-28`}>
			{menuElements.length > 0 &&
				menuElements.map((val, ind) => (
					<MenuElement /* color={isMobile ? "black" : undefined} */ title={val.name} onClicked={val.onClick} selected={selectedId === val.id} key={`menuElement${ind}`} />
				))}
		</div>
		// </MenuContainer>
	);
};

export default ListMenuElements;
