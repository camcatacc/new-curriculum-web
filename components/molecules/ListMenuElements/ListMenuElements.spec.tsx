// Modules
import React from "react";
import { fireEvent, render } from "@testing-library/react";

// Elements
import ListMenuElements from "components/molecules/ListMenuElements/ListMenuElements";

// Definitions
import type { ListMenuElementsProps } from "components/molecules/ListMenuElements/ListMenuElements";

// Mocks
const onClickMock = jest.fn();

// Default Props
const defaultProps: ListMenuElementsProps = {
	menuElements: [{ name: "Home", onClick: onClickMock, id: "home>" }],
	selectedId: "home"
};

// Tests
describe("ListMenuElements", () => {
	beforeAll(() => {
		Object.defineProperty(window, "matchMedia", {
			writable: true,
			value: jest.fn().mockImplementation((query) => ({
				matches: false,
				media: query,
				onchange: null,
				addListener: jest.fn(), // Deprecated
				removeListener: jest.fn(), // Deprecated
				addEventListener: jest.fn(),
				removeEventListener: jest.fn(),
				dispatchEvent: jest.fn()
			}))
		});
	});

	it("Renders as expected", () => {
		const { container } = render(<ListMenuElements {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
	it("Calls onClick function", () => {
		const { getByText } = render(<ListMenuElements {...defaultProps} />);
		const home = getByText("Home");
		fireEvent.click(home);

		expect(onClickMock).toHaveBeenCalled();
	});
});
