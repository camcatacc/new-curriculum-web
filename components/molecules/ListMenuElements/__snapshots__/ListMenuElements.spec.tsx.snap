// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenuElements Renders as expected 1`] = `
<div>
  <div
    class="undefined flex-row flex gap-28"
  >
    <p
      class="MuiTypography-root MuiTypography-body2"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-bdvvtL nvKmn cursor-pointer font-regular uppercase MuiTypography-colorTextSecondary"
      >
        Home
      </a>
    </p>
  </div>
</div>
`;
